### ISSUE TITLE
ensemble scheduler reshape not working as expected

### ISSUE THREAD

**Issue by mhunsake** [NVIDIA/tensorrt-inference-server#1055]
**Description**
Same use case as #968.   

Model A max batch size 16 -> Model B max batch size 0 -> model C max batch size 16.  Model A output is getting incorrectly reshaped prior to validating match with Model B input.  

The issue isn't just with the batch size (-1 vs 1), but it also reshaped the image tensor. Output of Model A is 1280,1920,3, but this error is showing that Model B received a tensor with shape 1920,3,1280.   Does the ensemble scheduler reshape make any assumptions regarding NHWC vs NCHW?

```tensorrtserver.api.InferenceServerException: [inference:0 14] in ensemble 'ensemble_bristol4', unexpected shape for input 'INPUT' for model 'custom_mkh4'. Expected [-1,1280,1920,3], got [1,1920,3,1280]```

Relevant config.pbtx below: 
```
# MODEL A ###############################################

max_batch_size: 16
input [
  {
    name: "INPUT"
    data_type: TYPE_STRING
    dims: [ 1 ]
  }
]
output [
  {
    name: "OUTPUT"
    data_type: TYPE_FP32
    dims: [ 1280, 1920, 3 ]
  }
]

# MODEL B ###############################################

max_batch_size: 0
input [
  {
    name: "INPUT"
    data_type: TYPE_FP32
    dims: [ -1, 1280, 1920, 3 ]
  }
]
output [
  {
    name: "OUTPUT"
    data_type: TYPE_FP32
    dims: [ -1, 299, 299, 3 ]
  }
]

# MODEL C ###############################################

max_batch_size: 16
input [
   {
      name: 'inception_v3_input'
      data_type: TYPE_FP32
      format: FORMAT_NHWC
      dims: [ 299, 299, 3]
   }
]
output [
   {
      name: 'dense'
      data_type: TYPE_FP32
      dims: [ 37 ]
      label_filename: 'labels.txt'
   }
]
```

**TRTIS Information**
nvcr.io/nvidia/tensorrtserver:19-12-py3

**Expected behavior**
I expect that the ensemble scheduler will just handle the batch size of -1 and will not swap the other dimensions of the input tensor.

**Comment by GuanLuo** [comment]
Can you also provide the configuration of the ensemble?

**Comment by mhunsake** [comment]
```
name: "ensemble_test"
platform: "ensemble"
max_batch_size: 16
input [
  {
    name: "INPUT"
    data_type: TYPE_STRING
    dims: [ 1 ]
  }
]
output [
  {
    name: "OUTPUT"
    data_type: TYPE_FP32
    dims: [ 37 ]
  }
]
ensemble_scheduling {
  step [
    {
      model_name: "image_string_to_mat"
      model_version: -1
      input_map {
        key: "INPUT"
        value: "INPUT"
      }
      output_map {
        key: "OUTPUT"
        value: "the_cv_mat"
      }
    },
    {
      model_name: "custom_mkh4"
      model_version: -1
      input_map {
        key: "INPUT"
        value: "the_cv_mat"
      }
      output_map {
        key: "OUTPUT"
        value: "preprocessed_image"
      }
    },
    {
      model_name: "my_inception_model"
      model_version: -1
      input_map {
        key: 'inception_v3_input'
        value: "preprocessed_image"
      }
      output_map {
        key: 'dense'
        value: "OUTPUT"
      }
    }
  ]
}
```

**Comment by mhunsake** [comment]
For completeness - I uploaded the full config.pbtxt files for each individual model plus the ensemble.

[the_configs.txt](https://github.com/NVIDIA/tensorrt-inference-server/files/4105847/the_configs.txt)

**Comment by GuanLuo** [comment]
There is a bug in the ensemble reshape function. Created PR #1058 to fix it.

**Comment by mhunsake** [comment]
Thanks!  I've downloaded your updated code to test.

**Comment by mhunsake** [comment]
@GuanLuo  I'm following instructions for Incremental Builds with Docker, but it is taking a really long time to build tensorrtserver_build. It's been running for over 90 minutes and still isn't done. Is this to be expected?

**Comment by GuanLuo** [comment]
I am suspecting your docker build also builds TensorFlow and other framework libraries.. That is normal for first time docker build as there aren't cache for those steps.

There is an workaround but I haven't tried this myself: you should be able to remove those stages for framework libraries and use the ones shipped with the NGC container.

**Comment by mhunsake** [comment]
I got it by using the Dockerfile as reference.  I started with nvcr.io/nvidia/tensorrtserver:19.12-py3 container, apt-get install necessary packages, and then built this new code.

tested my ensemble and it is working as expected.  Thanks for the fix!!

**Comment by GuanLuo** [comment]
Fixed by #1058. Closing the issue.
